jlm/llvm/opt/alias-analyses/RegionAwareModRefSummarizer.cpp

```diff
@@ -535,8 +337,115 @@ class RegionAwareModRefSummary final : public ModRefSummary
 
 private:
   const PointsToGraph & PointsToGraph_;
-  RegionSummaryMap RegionSummaries_;
-  CallSummaryMap CallSummaries_;
+
+  /**
+   * All sets of ModRef information in the summary
+   */
+  std::vector<ModRefSet> ModRefSets_;
+
+  /**
+   * Map from nodes that have memory side effects, to their ModRefSet.
+   * Includes nodes like loads, stores, memcpy, free and calls.
+   * Also includes structural nodes like gamma, theta and lambda.
+   */
+  std::unordered_map<const rvsdg::Node *, ModRefSetIndex> NodeMap_;
+};
+
+/**
+ * Struct holding temporary data used during the creation of a single mod/ref summary
+ */
+struct RegionAwareModRefSummarizer::Context
```

Comment:
I would rather like if the dependencies between the individual stages in the pipeline are made clear by handing over only the necessary information for each stage, instead of handing over the context and then documenting here what is not modified when etc.

This does not need to be done in this PR.